cmd/mdatagen/metrics.tmpl

```diff
@@ -19,7 +19,16 @@ import (
 
 // MetricSettings provides common settings for a particular metric.
 type MetricSettings struct {
-	Enabled bool `mapstructure:"enabled"`
+	defaultEnabled bool
+	Enabled        *bool `mapstructure:"enabled"`
+}
+
+// IsEnabled returns a boolean flag defining if the metric is enabled by looking at user settings and the default value.
+func (ms MetricSettings) IsEnabled() bool {
```

Comment:
Should this also add an `IsUserProvided() bool` helper?  It's not hard to check `ms.Enabled != nil` but having a common way of doing it is helpful.

Reply:
Makes sense. I also realized that using pointers is not the best way. Let me try a custom unmarshaler instead

Reply:
Custom unmarshaler worked pretty well. @Aneurysm9 PTAL